logic/transaction.js

```diff
@@ -65,1057 +51,1165 @@ __private.types = {};
+				return setImmediate(cb, 'Transaction#countById error');
+			});
 	}
 
-	// if (!this.ready(trs, sender)) {
-	// 	return setImmediate(cb, 'Transaction is not ready: ' + trs.id);
-	// }
-
-	// Check sender
-	if (!sender) {
-		return setImmediate(cb, 'Missing sender');
+	/**
+	 * Description of the function.
+	 *
+	 * @param {transaction} transaction
+	 * @param {function} cb
+	 * @returns {SetImmediate} error
+	 * @todo Add description for the params
+	 */
+	checkConfirmed(transaction, cb) {
+		this.countById(transaction, (err, count) => {
+			if (err) {
+				return setImmediate(cb, err);
+			} else if (count > 0) {
+				return setImmediate(
+					cb,
+					`Transaction is already confirmed: ${transaction.id}`
+				);
+			}
+			return setImmediate(cb);
+		});
 	}
 
-	// Get transaction id
-	var txId;
-
-	try {
-		txId = this.getId(transaction);
-	} catch (e) {
-		this.scope.logger.error(e.stack);
-		return setImmediate(cb, 'Failed to get transaction id');
+	/**
+	 * Checks if balance is less than amount for sender.
+	 *
+	 * @param {number} amount
+	 * @param {number} balance
+	 * @param {transaction} transaction
+	 * @param {account} sender
+	 * @returns {Object} With exceeded boolean and error: address, balance
+	 * @todo Add description for the params
+	 */
+	checkBalance(amount, balance, transaction, sender) {
+		const exceededBalance = new bignum(sender[balance].toString()).lessThan(
+			amount
+		);
+		const exceeded =
+			transaction.blockId !== this.scope.genesisblock.block.id &&
+			exceededBalance;
+
+		return {
+			exceeded,
+			error: exceeded
+				? [
+						'Account does not have enough LSK:',
+						sender.address,
+						'balance:',
+						new bignum(sender[balance].toString() || '0').div(Math.pow(10, 8)),
+					].join(' ')
+				: null,
+		};
 	}
 
-	// Check transaction id
-	if (transaction.id && transaction.id !== txId) {
-		return setImmediate(cb, 'Invalid transaction id');
-	}
-	transaction.id = txId;
+	/**
+	 * Validates parameters.
+	 * Calls `process` based on transaction type (see privateTypes).
+	 *
+	 * @see {@link privateTypes}
+	 * @param {transaction} transaction
+	 * @param {account} sender
+	 * @param {account} requester
+	 * @param {function} cb
+	 * @returns {SetImmediate} error, transaction
+	 * @todo Add description for the params
+	 */
+	process(transaction, sender, requester, cb, tx) {
+		if (typeof requester === 'function') {
+			cb = requester;
+		}
 
-	// Equalize sender address
-	transaction.senderId = sender.address;
+		// Check transaction type
+		if (!__private.types[transaction.type]) {
+			return setImmediate(cb, `Unknown transaction type ${transaction.type}`);
+		}
 
-	// Call process on transaction type
-	__private.types[transaction.type].process.call(
-		this,
-		transaction,
-		sender,
-		(err, transaction) => {
-			if (err) {
-				return setImmediate(cb, err);
-			}
-			return setImmediate(cb, null, transaction);
-		},
-		tx
-	);
-};
+		// if (!this.ready(trs, sender)) {
```

Comment:
Is this by mistake?

Reply:
this change was there before, so I have no clue why it was commented, let's discuss if we want to make any change here or remove it if not required.